src/main/java/org/jabref/gui/preftabs/GroupsPrefsTab.java

```diff
@@ -114,7 +110,14 @@ public void setValues() {
         groupingField.setText(prefs.get(JabRefPreferences.GROUPS_DEFAULT_FIELD));
         keywordSeparator.setText(prefs.get(JabRefPreferences.KEYWORD_SEPARATOR));
         autoAssignGroup.setSelected(prefs.getBoolean(JabRefPreferences.AUTO_ASSIGN_GROUP));
-        multiSelectionModeIntersection.setSelected(prefs.getBoolean(JabRefPreferences.GROUP_INTERSECT_SELECTIONS));
+
+        boolean intersectionSelected = prefs.getBoolean(JabRefPreferences.GROUP_INTERSECT_SELECTIONS);
+        //Looks odd, but the prefs store true for intersection and false for union
+        if (intersectionSelected) {
```

Comment:
Since this is hard to understand, I would propose to add a better named method to `JabrefPreferences`. Moreover, the code is probably easier to understand if you add an enum for the different modes.

Reply:
Now introduced an enum with methods for setting/get the pref value

Reply:
Personally, I would separate the group mode enum from the icon. The mode is something that can (in principle) be defined and used in the model/logic package thus I wouldn't mix it with something gui related.

Reply:
Well, however, this is purely a gui thing.  My immediate thoughts were also about storing that in model, but it would provide any additional value. And this makes it easier to directly use the icon,